inference-engine/src/transformations/src/transformations/common_optimizations/mul_conv_fusion.cpp

```diff
@@ -0,0 +1,269 @@
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        if (is_dequantization_subgraph(pattern_to_output.at(mul_pattern)))
+            return false;
+
+        const auto& weights = pattern_to_output.at(weights_pattern);
+        const auto& mul_const = pattern_to_output.at(mul_const_pattern);
+
+        const auto& weights_shape = weights.get_shape();
+        const auto& mul_const_shape = mul_const.get_shape();
+        if (op::util::check_for_broadcast(weights_shape, mul_const_shape) ||
+            (weights_shape.size() == mul_const_shape.size() && mul_const_shape[0] != 1)) {
+            return false;
+        }
+
+        auto weights_multiply = std::make_shared<opset8::Multiply>(weights, mul_const);
+        std::shared_ptr<Node> new_weights = get_constant_from_source(weights_multiply);
+        if (!new_weights)
+            new_weights = weights_multiply;
+
+        const auto& input = pattern_to_output.at(input_pattern);
+        const auto& conv = pattern_to_output.at(conv_pattern).get_node_shared_ptr();
+
+        auto new_conv = conv->clone_with_new_inputs({input, new_weights});
+        new_conv->set_friendly_name(conv->get_friendly_name());
+        copy_runtime_info({conv, pattern_to_output.at(mul_pattern).get_node_shared_ptr()},
+                          {new_weights, new_conv});
+        replace_node(conv, new_conv);
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(conv_pattern, matcher_name);
+    register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::MultiplyGroupConvolutionFusion, "MultiplyGroupConvolutionFusion", 0);
+
+ngraph::pass::MultiplyGroupConvolutionFusion::MultiplyGroupConvolutionFusion() {
+    MATCHER_SCOPE(MultiplyGroupConvolutionFusion);
+    auto input_pattern = pattern::any_input();
+    auto mul_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto mul_pattern = ngraph::pattern::wrap_type<opset8::Multiply>({input_pattern, mul_const_pattern}, pattern::consumers_count(1));
+    auto weights_pattern = ngraph::pattern::any_input(pattern::has_static_shape());
+    auto conv_pattern = ngraph::pattern::wrap_type<opset8::GroupConvolution>({mul_pattern, weights_pattern});
+
+    matcher_pass_callback callback = [=](pattern::Matcher & m) -> bool {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        if (is_dequantization_subgraph(pattern_to_output.at(mul_pattern)))
+            return false;
+
+        const auto& weights = pattern_to_output.at(weights_pattern);
+        std::shared_ptr<Node> mul_const = pattern_to_output.at(mul_const_pattern).get_node_shared_ptr();
+
+        const auto& weights_shape = weights.get_shape();
+        auto mul_const_shape = mul_const->get_shape();
+        if (shape_size(mul_const->get_shape()) > 1) {
+            auto mul_const_shape = mul_const->get_shape();
+            if (weights_shape.size() - mul_const_shape.size() > 1)
```

Comment:
> It will catch an overflow if mul_const_shape.size() > weights_shape.size().

validate_and_infer_types in GroupConvolution checks that weights rank == input rank + 1 so there should be no underflow here.

> And why are you not broadcasting mul_const_shape to the weights_shape size?

I was considering that, but it's easier to explain following check "mul_const_shape[0] != 1" later if I extend mul_const rank to "weights_shape.size() - 1".